Foundation/NSHTTPCookieStorage.swift

```diff
@@ -47,7 +47,18 @@ open class HTTPCookieStorage: NSObject {
         allCookies = [:]
         cookieAcceptPolicy = .always
         super.init()
-        cookieFilePath = filePath(path: _CFXDGCreateDataHomePath()._swiftObject, fileName: "/.cookies." + cookieStorageName)
+        //let bundleName = Bundle.main.object(forInfoDictionaryKey: "CFBundleDisplayName") as! String
+        //TODO: Above api returns nil on Linux when this code is introduced. Hence, using the below workaround.
+        //Needs replacement once fixed.
+        let bundlePath = Bundle.main.bundlePath
+        var bundlePathComponents = bundlePath.components(separatedBy: "/")
+        let bundleDesc: String = bundlePathComponents[bundlePathComponents.count - 1]
+        bundleName = bundleDesc
+        if let range = bundleDesc.range(of: ".", options: String.CompareOptions.backwards, range: nil, locale: nil) {
```

Comment:
There's a `deletingPathExtension` property on NSString you can use if that's what you're trying to do here.

https://developer.apple.com/reference/foundation/nsstring/1418214-deletingpathextension

You don't do this in the code below that also retrieves the bundle name though.